python/lsst/cp/pipe/astierCovFitParameters.py

```diff
@@ -0,0 +1,282 @@
+#  Code taken from Pierre Astier's bfptc gitHub repository, where the following
+#  is written:
+#  "I don't know how wrote that. It belonged to "saunerie", developed
+#  at LPNHE (Paris). Just copied here."
+
+"""
+This module contains utility classes to handle parameters in linear and
+non-linear least square fits implemented in linearmodels and
+nonlinearmodels. It provide 2 main features:
+
+- `StructArray`: a derivative of numpy class ndarray to manage large
+vectors organized into named subgroups.
+
+- `FitParameters` : a class to manage large parameter vectors.  It
+allows to easily fix/release specific parameters or entire subgroups,
+and remap the remaining free parameters into a contiguous vector.
+"""
+
+import numpy as np
+
+
+class Structure(object):
```

Comment:
I’m not sure if changing "class Structure(object)" for a "pipeBase.Struct" will preserve the methods and properties of "StructArray(np.ndarray)" (based on "Structure") and "FitParameters" (based on "StructArray") in "astierCovFitParameters.py". In the end, "FitParameters" is only used in "self.params" for the fits in "astierCovPtcFit.py". I chose to leave the whole astierCovFitParameters.py to avoid breaking things.